### ISSUE TITLE
Regression issue - hidden fields can not be made visable

### ISSUE THREAD

**Issue by ceblanda** [xamarin/Xamarin.Forms#1736]
### Description
Running on Android, a page in a MasterDetail Detail that contains hidden elements in a scroll.  If you change the elements to be visible they do not show.

### Steps to Reproduce

1. Run the attached project.  Notice that the first view has a switch.  Switch the switch and the hidden elements will become visable.
2. Open the Master window and choose Page1.  This is the same exact page, same code, but now the switch does not make the elements appear.
3. Open the Master window and choose Page2.  This is an identical pages but does not contain a ScrollView.  Note that the switch does work here.
4. Using Nuget, change the version of Forms to 2.3.4.270.  Clean and Rebuild All.  Now all 3 pages behave the same.

### Expected Behavior
The hidden elements should become visible after the IsVisible property has been set to Visible

### Actual Behavior
The hidden elements do not appear.

### Basic Information

- Version with issue:
- Last known good version: 2.3.4.270
- IDE:VS2017

### Reproduction Link

[HeaderTest.zip](https://github.com/xamarin/Xamarin.Forms/files/1675405/HeaderTest.zip)

**Comment by PureWeen** [comment]
Repro confirms issues

Tested with latest Xam Forms / Android libraries as well.
iOS works hides and shows the field correctly 
Android does not

Thank you for the reproduction

**Comment by JacobChrist** [comment]
What happens if you rotate the screen, does the switch start working?

**Comment by JacobChrist** [comment]
I think I have a related problem.  I'm changing the text of a label in a Detail.  The label will not resize as expected (that is at all).  However, if I rotate the screen then the label will resize as expected.  The Detail page works as expected if used outside of the MasterDetail relationship.  Also, after I found your post I took my label out of the ScrollView and it now works in the MasterDetail relationship.  I also have similar problems with buttons in a ScrollView in a MasterDetail relationhip.
